### DISCUSSION TITLE
Chain Load an ipxe.efi from GCE Cloud Configured ipxe.usb Image

### DISCUSSION THREAD

**Discussion by erl-hpe** [ipxe/ipxe#338] [Q&A] [answered]
I have a sort of weird use case in which I am booting a GCE instance into iPXE using the GCE flavor of the `cloud` configuration of iPXE burned to a USB image and written to a GCE disk image.  So far, nothing too special, but from that iPXE I need to chain-load a specially configured ipxe.efi from an HTTP server running on another GCE instance and then let the chain-loaded ipxe.efi take the boot from there.  The reasons for doing this are complicated and have to do with simulating a hardware compute cluster, and are not really relevant here.  The problem I am running into is that the ipxe.usb that I get when I build using:
```
make bin/ipxe.usb CONFIG=cloud EMBED=config/cloud/gce.ipxe
```
does not contain support for executing EFI images, and, for the life of me, I can't find the knobs I need to turn to add that capability to ipxe.usb.  Is there anyone out there who can point me in the right direction?

**Comment by mcb30** [comment, marked as answer]
UEFI and BIOS are different firmwares.  There is no way to boot an EFI executable on a non-EFI platform such as GCE.  You will need to chainload into a BIOS build of iPXE (e.g. `bin/ipxe.pxe`) instead of `ipxe.efi`.

**Reply by erl-hpe** [reply]
Ah.  Okay, thanks.  I was wondering if that was where this was going to end up.  I think I can work with that.  Thanks for the quick response!

**Comment by sebaxakerhtc** [comment]
When i need to test any EFI files or installations of EFI OS's on non-EFI platform - I use Clover bootloader.
But better way is to use Legacy iPXE on legacy platform as @mcb30 said [here](https://github.com/ipxe/ipxe/discussions/338#discussioncomment-652383)

**Reply by erl-hpe** [reply]
Yes, I think legacy iPXE is going to work fine for me. I was able to get the workflow I need to work properly using the ipxe.usb boot image configured for gce and an ipxe.pxe configured for cloud, and the platform I am emulating is able to provide that with some minor changes that should make it more flexible in the long run anyway.  Thanks all for the help!
